src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -1462,6 +1462,18 @@ void os::print_memory_info(outputStream* st) {
   st->cr();
 }
 
+size_t os::get_rss() {
+  size_t result = 0;
+#ifdef __APPLE__
+  mach_task_basic_info info;
+  mach_msg_type_number_t count = MACH_TASK_BASIC_INFO_COUNT;
+  if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO, (task_info_t)&info, &count) == KERN_SUCCESS) {
+    result = info.resident_size;
+  }
+#endif // __APPLE__
```

Comment:
Hmmm so no general BSD support either ...

Reply:
It is possible on BSD (e.g. using [kvm_getprocs](https://man.freebsd.org/cgi/man.cgi?query=kvm_getprocs&sektion=3&n=1) ) - however, I am no BSD expert and have no system to build BSD on.

Note that in both cases - AIX and the BSDs - I am not better nor worse than other code that reads RSS, e.g. jfr_report_memory_info(). Which, btw, is a code unification possibility I plan on following up on.

Reply:
I don't like seeing disparity in platform support like this but understand we can't always supply the details for every port on the initial integration. I would like to see RFE's filed to have the missing details provided on those platforms so that the respective port maintainers  know there is something missing. I suspect that is not the case with `jfr_report_memory_info`.

Reply:
JBS is down for maintenance; will open bugs when its back up. 

Platform support is always an issue; BSD support, in particular, seems spotty. I am not even sure BSD is maintained by anyone in the head release.